### ISSUE TITLE
Error NU1004 disappears when building solution after enabling “RestoreLockedMode”

### ISSUE THREAD

**Issue by v-nancyj** [NuGet/Home#8973]
## Details about Problem

NuGet Version: Dev\5.5.0.6330
VS Version: Master\29623.15
OS Version: Windows-10-Enterprise-RS5

## Installation steps: 
* https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3341765&view=results

## Repro steps: 
1. Create above corresponding projects via VS. 
2. Install a few ‘PackageReference’ packages into the projects.
3. Set MSBuild property “RestorePackagesWithLockFile” to true in csproj file.
  `<RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>`
4. Do a restore.
5. Check it should have generated ‘packages.lock.json’ file alongside *.csproj file.
6. Set msbuild property “RestoreLockedMode” to “true”.
    `<RestoreLockedMode>true</RestoreLockedMode>`
7. Install another package.
8. It should still work fine and update ‘packages.lock.json’ file.
9. Manually add a new PackageReference item into csproj file.
```
<PackageReference Include="NUnit">
   <Version>3.12.0</Version>
 </PackageReference>
```
10. Perform restore.
11. Restore should fail with error “NU1004”
12. Build the solution.

## Expected:
The error was still showed on error list.

## Actual:
The error disappear on error list.

## Here is the record video for the repro steps:
![restore](https://user-images.githubusercontent.com/20632775/71407713-ccab5b80-2676-11ea-9195-da8a996d5488.gif)

## Notes: 
1. The issue repro on .NET Core project.
2. The issue repro on Master\29623.15 + NuGet Client Dev\5.5.0.6355.
3. The issue is not stable, the repro rate is almost 95% for us.

**Comment by zkat** [comment]
Hi @v-nancyj! Thanks for the bug report. Can you confirm whether this happens on any previous versions of 5.5, or in 5.4, if you tested those? I want to make sure it's a regression in 5.5.

**Comment by zkat** [comment]
I've confirmed that the expected behavior happens fine in 5.4, but not in 5.5.

**Comment by zkat** [comment]
I can no longer reproduce this with the latest build. I don't know what fixed it, but it got fixed. Maybe it was something with VS itself.

**Comment by v-nancyj** [comment]
@zkat Hi, Kat, During our investigation, we found the issue is not stable, the repro rate is almost 95% for us. I can still repro this on VS Master\29802.14 + NuGet Client Dev\5.5.0.6411 today, could you help to have a look and reopen it? Thanks!

**Comment by zkat** [comment]
@v-nancyj confirmed. I tried again a couple of times and it's definitely still there. I'll take a closer look now, thanks!

**Comment by zkat** [comment]
Update: This is not a regression during 16.5. I can repro it in 16.4.

The issue is that the [`ErrorListTableDataSource.ClearNuGetEntries()` method](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Common/ErrorListTableDataSource.cs#L109) isn't installed for the `build` action in VS. That gets [installed into the build event](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.Tools/OutputConsoleLogger.cs#L68) when [the PMUI is loaded](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L193).

So basically, the issue is not that this is getting cleared -- the issue is that we _want_ it to clear on build (or at least that's what we intended to do), but the clear event isn't installed unless you use the PMUI. I don't know what our actual desired behavior is, but I'm tempted to open a separate issue and close this specific one so we can discuss from there.

**Comment by zkat** [comment]
Moving this to the icebox -- I don't think this is high-priority enough for us to fix right now.

**Comment by cristinamanum** [comment]
The NU1004 is not added to the assets file. Visusl Studio version used for repo : Version 16.7.0 Preview 4.0 [30229.9.master]

Repo steps:
Some repro steps.

Create a PR project
Add a few dependencies & set RestorePackagesWithLockFile.
Edit a csproj manually adding a new PackageREference by editing the version 
Check if the NU1004 is in the assets file.

**Comment by nkolev92** [comment]
I think @cristinamanum's step explains why NU1004 disappears. 

For .NET Core projects, things need to be in the assets file to stay in the error list reliably, moving to backlog because this is a core behavior and we now understand the root cause.

**Comment by nkolev92** [comment]
Fix out in https://github.com/NuGet/NuGet.Client/pull/3816.
